src/wallet/wallet2.cpp

```diff
@@ -5478,6 +5478,13 @@ void wallet2::set_ring_database(const std::string &filename)
     m_ringdb.reset(new tools::ringdb(m_ring_database, epee::string_tools::pod_to_hex(get_block_hash(b))));
 }
 
+std::string wallet2::get_ring_database()
+{
+  if(m_ring_database.empty())
+    m_ring_database = get_default_ringdb_path();
```

Comment:
why not make it this way:
```
if(m_ring_database.empty())
  return get_default_ringdb_path();
else 
  return m_ring_database;
```

this would make it possible to mark the function as `const` which is natural and preferred for getters

Reply:
and even better to set `m_ring_database` to `get_default_ringdb_path()` during initialization. that'll let make the getter `const` and safe to use naked member in wallet2 nuts